Feature/better way to write junit xml

sbonacho opened the pull request:
escribe directamente un fichero junit.xml sin usar mocha..

antiguamente usando mocha el time no era el correcto puesto que se hacía ejecutando otra vez un test. No era correcto.

Commit:
- Final Check of the execution
- Write junit xml without mocha
- take the time of every stage of a shot

Commit:
- count of errors and faliures
- time spend
- timestamp

biaggi commented on lib/finalCheck.js:
```diff
@@ -0,0 +1,107 @@
+    };
+
+    var writeJunitXml = function(junit){
+        var file = path.join(config.junitDir, "pisco-unit.xml");
```
Estaría bien que pisco-unit.xml fuera una constante

sbonacho replied:
hecho!

biaggi commented on lib/finalCheck.js:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    var shot2Testsuite = function(junit, report){
+        var testsuite = junitGen.testsuite();
```
Si esto es un api para generar xml, igual debería tener un facade por delante por si en algún momento la queremos cambiar

sbonacho replied:
Realmente es el uso de un api para escribir concretamente un fichero junit a partir de un shot y un straw

biaggi commented on lib/shot.js:
```diff
@@ -37,6 +37,7 @@ runner.on("end",function(){
 */
 
 Shot.prototype.do = function(stage){
+    this.init = new Date().getTime();
```
Quizás aqui se debería estar usando moment

sbonacho replied:
Realmente lo que queremos es obtener los milisegundos para calcular el tiempo transcurrido. Creo que moment no es necesario para eso, no? por lo que he visto es para formatear fechas.

biaggi commented on lib/sour.js:
```diff
@@ -17,6 +18,14 @@ var docs = require("./docs"),
+        try{
+            return finalCheck.run(init);
+        }catch (e){
+            console.error(e);
```
Esto igual debería ir al logger

sbonacho replied:
hecho!

Commit:
- Delete some comments
- Junit file name in configuration as junitPiscoFile
- Fix console output to logger

biaggi commented:
+1
